add (redux logic): Implement redux to project

untael opened the pull request.

untael committed:
Merge branch 'master' into dev

untael committed:
wip (redux logic): Try to implement redux #2

untael committed:
add (redux logic): Implement redux to project

iTechJ commented on .eslintrc.js:
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    "extends": "google"
```
я бы добавил что-нибудь специфичное для реакта. Как пример конфиг с проекта https://gist.github.com/iTechJ/e20dd6bfcbdb3db38c6ca721989ed392

iTechJ commented on src/components/ContactCreateForm/ContactCreateForm.js:
```diff
@@ -12,38 +9,18 @@ class ContactCreateForm extends React.Component {
-    //   })
-    //   console.log('state', this.state.contact)
-    // }
     if (value !== '') {
```
if (!!value)

iTechJ commented on src/components/ContactExplorer/ContactExplorer.js:
```diff
@@ -160,47 +61,31 @@ class ContactExplorer extends React.Component {
-              updateList={this.updateList}
-            />
+            <ContactCreateForm/>
           </div>) : (null)
```
`this.state.showEdit && <Component/>`

iTechJ commented on src/containers/ContactListItem.js:
```diff
@@ -1,13 +1,15 @@
+  showDisplayPanel: (contact) => dispatch(showDisplay(contact)),
+  showEditPanel: (contact) => dispatch(showEdit(contact)),
+  deleteContact: (contactId) => dispatch(deleteContact(contactId)),
 })
```
https://stackoverflow.com/a/48629997 То что я говорил, можно упростить запись

iTechJ commented on src/index.js:
```diff
@@ -1,18 +1,19 @@
 
 const rootEl = document.getElementById('root')
-const store = createStore(rootReducer)
+const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), applyMiddleware(thunk))
```
Лучше вынести это в отдельный файл

iTechJ reviewed (approved):
Замечания/предложения только по мелочи. Поработай над асинхронными экшенами и error handling.

untael committed:
refactor (components): Wizard instead full form, add loader

Now list can have only 10 items on start, and loader can load more items if need
